src/renderer/base/renderer.cpp

```diff
@@ -772,14 +772,14 @@ void Renderer::_PaintBufferOutput(_In_ IRenderEngine* const pEngine)
                 scratch.CopyFrom(r);
                 rowBackup = &scratch;
 
-                std::wstring_view text{ _pData->activeComposition.text };
+                std::wstring_view text{ _pData->GetActiveComposition().text };

```

Comment:
Could you add `const auto& activeComposition = _pData->GetActiveComposition()` here and elsewhere below so that we don't need to call the virtual function multiple times?